ql/termstructures/yield/compositezeroyieldstructure.hpp

```diff
@@ -0,0 +1,157 @@
+      private:
+        Handle<YieldTermStructure> curve1_;
+        Handle<YieldTermStructure> curve2_;
+        BinaryFunction f_;
+        Compounding comp_;
+        Frequency freq_;
+        DayCounter dc_;
+    };
+
+    // inline definitions
+
+    template <class BinaryFunction>
+    inline CompositeZeroYieldStructure<BinaryFunction>::CompositeZeroYieldStructure(
+        const Handle<YieldTermStructure>& h1, 
+        const Handle<YieldTermStructure>& h2, 
+        const BinaryFunction& f,
+        Compounding comp, 
+        Frequency freq, 
+        const DayCounter& dc)
+    : curve1_(h1), curve2_(h2), f_(f), comp_(comp), freq_(freq), dc_(dc) {
+        if (!curve1_.empty() && !curve2_.empty())
+            enableExtrapolation(curve1_->allowsExtrapolation() && curve2_->allowsExtrapolation());
+
+        registerWith(curve1_);
+        registerWith(curve2_);
+    }
+
+    template <class BinaryFunction>
+    inline DayCounter CompositeZeroYieldStructure<BinaryFunction>::dayCounter() const {
+        return curve1_->dayCounter();
+    }
+
+    template <class BinaryFunction>
+    inline Calendar CompositeZeroYieldStructure<BinaryFunction>::calendar() const {
+        return curve1_->calendar();
+    }
+
+    template <class BinaryFunction>
+    inline Natural CompositeZeroYieldStructure<BinaryFunction>::settlementDays() const {
+        return curve1_->settlementDays();
+    }
+
+    template <class BinaryFunction>
+    inline const Date& CompositeZeroYieldStructure<BinaryFunction>::referenceDate() const {
+        return curve1_->referenceDate();
+    }
+
+    template <class BinaryFunction>
+    inline Date CompositeZeroYieldStructure<BinaryFunction>::maxDate() const {
+        return curve1_->maxDate();
+    }
+
+    template <class BinaryFunction>
+    inline Time CompositeZeroYieldStructure<BinaryFunction>::maxTime() const {
+        return curve1_->maxTime();
+    }
+
+    template <class BinaryFunction>
+    inline void CompositeZeroYieldStructure<BinaryFunction>::update() {
+        if (!curve1_.empty() && !curve2_.empty()) {
+            YieldTermStructure::update();
+            enableExtrapolation(curve1_->allowsExtrapolation() && curve2_->allowsExtrapolation());
+        }
+        else {
+            /* The implementation inherited from YieldTermStructure
+            asks for our reference date, which we don't have since
+            the original curve is still not set. Therefore, we skip
+            over that and just call the base-class behavior. */
+            TermStructure::update();
+        }
+    }
+
+    template <class BinaryFunction>
+    inline Rate CompositeZeroYieldStructure<BinaryFunction>::zeroYieldImpl(Time t) const {
+        Rate zeroRate1 =
+            curve1_->zeroRate(t, comp_, freq_, true);
+
+        InterestRate zeroRate2 =
+            curve2_->zeroRate(t, comp_, freq_, true);
+
+        InterestRate compositeRate(f_(zeroRate1, zeroRate2), dc_, comp_, freq_);
```

Comment:
The two rates above use the day-count convention of their curves (which by construction is the same as the one of this curve), so you should use `dayCounter()` and not `dc_`.  This makes `dc_` unused, so you can remove it.